### PR TITLE
Replace deprecated `forcats::fct_explicit_na`

### LINKED ISSUE #811: replace deprecated `forcats::fct_explicit_na`
> [fct_explicit_na()](https://forcats.tidyverse.org/reference/fct_explicit_na.html) is deprecated in favour of [fct_na_value_to_level()](https://forcats.tidyverse.org/reference/fct_na_value_to_level.html).

https://forcats.tidyverse.org/news/index.html#:~:text=fct_explicit_na()%20is%20deprecated%20in%20favour%20of%20fct_na_value_to_level().

Please don't forget to specify appropriate (i.e. minimal and fully compatible) version of `forcats` in the DESCRIPTION file.

This makes the tests output almost not readable. There are a lot of warnings like this:
```
  ✔ |   2    19 | compare_variables [0.4s]                                        
  ────────────────────────────────────────────────────────────────────────────────
  Warning ('test-compare_variables.R:32'): s_compare for factor handles explicit NAs as expected
  `fct_explicit_na()` was deprecated in forcats 1.0.0.
  i Please use `fct_na_value_to_level()` instead.
  Backtrace:
   1. tern::explicit_na(factor(c("a", "a", "b", "c", "a", NA, NA)))
        at test-compare_variables.R:32:2
   2. forcats::fct_explicit_na(x, label)
  
  Warning ('test-compare_variables.R:33'): s_compare for factor handles explicit NAs as expected
  `fct_explicit_na()` was deprecated in forcats 1.0.0.
  i Please use `fct_na_value_to_level()` instead.
  Backtrace:
   1. tern::explicit_na(factor(c("a", "b", "c", NA)))
        at test-compare_variables.R:33:2
   2. forcats::fct_explicit_na(x, label)
```

### PR DESCRIPTION
**Pull request by edelarua**
Closes #811

### CODE REVIEW

File: tests/testthat/_snaps/summarize_num_patients.md [hunk 1]
```diff
@@ -92,7 +92,7 @@
       ——————————————————————————————————————————————————————————————————
       Number of patients with at least one event   3 (60.0%)   3 (75.0%)
       Number of events                                 4           4    
-       (n)                                             3           3    
+      (n)                                              3           3
```
Comment (Melkiades) [inline tests/testthat/_snaps/summarize_num_patients.md:95]: why does this happen? could it be due to the recent indentation fix or is it concerning this PR?

File: NEWS.md [hunk 2]
```diff
@@ -50,6 +50,7 @@
 * Created separate documentation files for functions in different sections of pkgdown reference.
 * Created separate `.R` files for logistic regression and cox regression helper functions.
 * Removed all template tests from `tern`. These have been relocated to internal repo `scda.test`.
```
Comment (Melkiades) [inline NEWS.md:52]: So annoying these grammar complainings ahah

File: tests/testthat/_snaps/summarize_num_patients.md (hunk 1 above)
Comment (edelarua) [reply tests/testthat/_snaps/summarize_num_patients.md:95]: Not from changes in this PR. Probably from the indentation fix I would assume.

Comment (Melkiades) [reply tests/testthat/_snaps/summarize_num_patients.md:95]: I will double check this

File: NEWS.md (hunk 2 above)
Comment (edelarua) [reply NEWS.md]: Yeah, should be fixed now (I hope).

**Comment by Melkiades** [comment]
@cicdguy, we think the Docker image is missing the most recent version of `forcats` that is needed for this. Can we have it at 1.0.0 ? Also, we were wondering how to check grammar locally, so we are faster at writing texts

**Comment by edelarua** [comment]
> @cicdguy, we think the Docker image is missing the most recent version of `forcats` that is needed for this. Can we have it at 1.0.0 ?

@Melkiades Arek is rebuilding the ci-image to resolve the `forcats` version issue.

**Comment by edelarua** [comment]
@Melkiades versioning is fixed but I'm not able to reproduce the `g_lineplot` warnings that are causing test failures. I'm also not sure where these errors would stem from since `g_lineplot` doesn't seem to use any of the functions that were touched in this PR. Can you reproduce the failed tests?

**Comment by Melkiades** [comment]
I can not reproduce the warnings locally. I am updating everything and see if something changes

**Comment by Melkiades** [comment]
Lgtm anyway @edelarua! Thank you 👍

**Review by Melkiades — APPROVED** [review]
Lgtm anyway @edelarua! Thank you 👍

**Comment by edelarua** [comment]
@Melkiades I was not able to reproduce either. We could remove these tests for now (they were recently added) and look into this later.

**Comment by edelarua** [comment]
@shajoezhu This PR will fix `tern` for now and unblock issues in downstream dependencies (including `chevron` which is currently failing because of this deprecated function), but I see that you have added the blocked label. Should we still hold off on merging this?

**Comment by shajoezhu** [comment]
Hi @edelarua , yes please. Let's block this for one more day. I will follow up in the office with Liming, and we test it.

**Comment by shajoezhu** [comment]
Hi @edelarua , yes please. Let's block this for one more day. I will follow up in the office with Liming, and we test it.

**Comment by clarkliming** [comment]
works fine for downstread side 👍

**Pull request merged**